services/services.go

```diff
@@ -684,70 +677,118 @@ func getIndexPageData() (*types.IndexPageData, error) {
 		return nil, fmt.Errorf("error retrieving index block data: %v", err)
 	}
 
-	blocksMap := make(map[uint64]*types.IndexPageDataBlocks)
-	for _, block := range blocks {
-		if blocksMap[block.Slot] == nil || len(block.BlockRoot) > len(blocksMap[block.Slot].BlockRoot) {
-			blocksMap[block.Slot] = block
+	// keep slots with higher blockroot length
+	slotsMap := make(map[uint64]*types.IndexPageDataBlocks)
+	for _, slot := range slots {
+		if slot.Slot > (slots[0].Slot-10) && (slotsMap[slot.Slot] == nil || len(slot.BlockRoot) > len(slotsMap[slot.Slot].BlockRoot)) {
+			slotsMap[slot.Slot] = slot
 		}
 	}
-	blocks = make([]*types.IndexPageDataBlocks, 0, len(blocks))
-	for _, b := range blocksMap {
-		blocks = append(blocks, b)
-	}
-	sort.Slice(blocks, func(i, j int) bool {
-		return blocks[i].Slot > blocks[j].Slot
-	})
-	data.Blocks = blocks
 
-	if len(data.Blocks) > 15 {
-		data.Blocks = data.Blocks[:15]
-	}
-
-	for _, block := range data.Blocks {
-		block.StatusFormatted = utils.FormatBlockStatus(block.Status)
-		block.ProposerFormatted = utils.FormatValidatorWithName(block.Proposer, block.ProposerName)
-		block.BlockRootFormatted = fmt.Sprintf("%x", block.BlockRoot)
+	// set block struct values
+	for _, slot := range slotsMap {
+		slot.StatusFormatted = utils.FormatBlockStatusStyle(slot.Status)
+		slot.ProposerFormatted = utils.FormatValidatorWithNameAlt(slot.Proposer, slot.ProposerName)
+		slot.ExecutionRewardFormatted = template.HTML("-")
+		slot.ExecutionRewardRecipient = template.HTML("-")
+		slot.Ts = utils.SlotToTime(slot.Slot)
 
-		if !epochsMap[block.Epoch] {
-			epochs = append(epochs, &types.IndexPageDataEpochs{
-				Epoch:                            block.Epoch,
-				Ts:                               utils.EpochToTime(block.Epoch),
+		if _, containsEpoch := epochsMap[slot.Epoch]; !containsEpoch {
```

Comment:
Why not stay with `if !epochsMap[slot.Epoch]`?

Furthermore, I don't really understand why epochs have to be added like this. Could you please tell me?

Reply:
The slots are retrieved from the database after the epochs, so it's possible that newer slots from a more recent epoch than the most recent one in the epoch query get exported to the database in the interim. In this case, I want to display that epoch on the page since the page now also shows slots of that epoch. That's why I add it to the epochsMap, rather than skipping it when calculating the MEV reward.

Reply:
Thank you for the explanation and also thank you for adding the comment in the sources => Resolved

Reply:
Many thanks for the explanation and for adding the comment in the sources.
One more thing though: The slotsMap gets converted into a slice later and some slots may be removed from there, see [HERE](https://github.com/gobitfly/eth2-beaconchain-explorer/pull/1718/files/08a6614eee2a63eeb96bd65716101a2de37f2578..a8bf44365f2fe8cce60ba64fc4d107fae40df585#diff-3b37e5382c8ce5b2244a5256b612ea8e965943bc02ec783669c2e679e69d6760R822).

I think it would be better to add missing epochs after the slots have been reduced. Otherwise it MIGHT happen that you add an epoch for a slot that is not shown in the end.